### PR TITLE
Add benchmark for add op.

### PR DESCRIPTION
**Pull request by kimishpatel**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40252 Add add_relu fusion pass to optimize_for_mobile.
* #39343 JIT pass for add relu fusion.
* #39342 Add fused add_relu op.
* #39341 [Vec256][neon] Add neon backend for vec256
* **#40059 Add benchmark for add op.**

Summary:
This benchmark is added specifically for mobile to see if compiler is
autovectorizing and thus we have no advantage of neon backend for vec256
for add op.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22055146](https://our.internmc.facebook.com/intern/diff/D22055146)

### CODE REVIEW

File: aten/src/ATen/benchmarks/tensor_add.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#include <benchmark/benchmark.h>
+
+static void tensor_add(benchmark::State& state) {
+  const size_t batchSize = static_cast<size_t>(state.range(0));
+  const size_t channels = static_cast<size_t>(state.range(1));
+
+  at::Tensor a = at::rand({batchSize, channels});
+  at::Tensor b = at::rand({batchSize, channels});
+  at::Tensor c;
+  for (auto _ : state) {
```
Comment (AshkanAliabadi) [inline aten/src/ATen/benchmarks/tensor_add.cpp:12]: I'm not sure whether this runs, for a benchmark, the risk of getting optimized out considering the contents of this loop are invariant.

Comment (kimishpatel) [reply aten/src/ATen/benchmarks/tensor_add.cpp:12]: You mean the entire loop is optimized away?
This is how you are supposed to use google benchmark it seems: https://github.com/google/benchmark. Also qnnpack benchmarks are similarly written.
Also the next PR in the stack has the result of running this benchmark. From that it does not seem to be case.

**Comment by AshkanAliabadi** [comment]
LGTM

File: CMakeLists.txt [hunk 2]
```diff
@@ -132,6 +132,7 @@ cmake_dependent_option(
     CAFFE2_USE_MSVC_STATIC_RUNTIME "Using MSVC static runtime libraries" ON
     "NOT BUILD_SHARED_LIBS" OFF)
 option(BUILD_TEST "Build C++ test binaries (need gtest and gbenchmark)" OFF)
+option(BUILD_MOBILE_BENCHMARKS "Build C++ test binaries for mobile (ARM) targets(need gtest and gbenchmark)" OFF)
```
Comment (dreiss) [inline CMakeLists.txt:135]: space before `(`

**Review by dreiss — APPROVED** [review]

**Pull request closed without merging**